core/example/longrange/ewald.cpp

```diff
@@ -103,163 +66,404 @@ double TEwald::compute( ParticleList &particles, double lx, double ly,
+                                        double d =
+                                            sqrt( dx * dx + dy * dy + dz * dz );
+                                        double contrib =
+                                            ( d <= r_max &&
+                                              std::abs( d ) >= 1e-12 )
+                                                ? 0.5 * q( i ) * q( j ) *
+                                                      erfc( alpha * d ) / d
+                                                : 0.0;
+                                        double f_fact =
+                                            ( d <= r_max &&
+                                              std::abs( d ) >= 1e-12 ) *
+                                            q( i ) * q( j ) *
+                                            ( 2.0 * sqrt( alpha / PI ) *
+                                                  exp( -alpha * d * d ) +
+                                              erfc( sqrt( alpha ) * d ) ) /
+                                            ( d * d +
+                                              ( std::abs( d ) <= 1e-12 ) );
+                                        Kokkos::atomic_add( &f( i, 0 ),
+                                                            f_fact * dx );
+                                        Kokkos::atomic_add( &f( i, 1 ),
+                                                            f_fact * dy );
+                                        Kokkos::atomic_add( &f( i, 2 ),
+                                                            f_fact * dz );
+                                        Kokkos::single(
+                                            Kokkos::PerThread( member ),
+                                            [&] { Ur_j += contrib; } );
+                                    }
+                        },
+                        Ur_inner );
+                    Kokkos::single( Kokkos::PerTeam( member ), [&] {
+                        p( i ) += Ur_inner;
+                        Ur_i += Ur_inner;
+                    } );
+                }
+            },
+            Ur );
+        Kokkos::fence();
+        auto end_time_r = std::chrono::high_resolution_clock::now();
+        auto elapsed_time_r = end_time_r - start_time_r;
+        auto ns_elapsed_r =
+            std::chrono::duration_cast<std::chrono::nanoseconds>(
+                elapsed_time_r );
+
+        std::cout << "real-space contribution: "
+                  << ( ns_elapsed_r.count() / 1000000000.0 ) << " s " << Ur
+                  << std::endl;
+    }
+    else
+    {
+
+        // In this branch a linked-cell list is used, which is using the Cabana
+        // implementation of linked cell neighbor list. To compute the
+        // real-space contribution the particles are sorted into cells of a
+        // given size and then for each cell the distances to particles in
+        // surrounding cells are computed. If the distance is smaller than the
```

Comment:
It seems the `Cabana::VerletList` could be used here instead. You simply set the neighborhood radius to be the cut-off distance and you will immediately get a list of all the particles with which you would compute interactions.

Reply:
yeah you could delete most of the code below and just do the force computation